config/task.go

```diff
@@ -353,6 +371,12 @@ func (c *TaskConfig) Validate() error {
 			"version within the Terraform driver block", *c.Name)
 	}
 
+	if c.TFCWorkspace != nil && !c.TFCWorkspace.isEmpty() {
```

Comment:
So is the idea that this would be removed for CTS Enterprise?

Maybe we handle switching this on/off similar to how we handle the dev license, where a global flag is set only if building with the enterprise build tag? Not sure what the best way to handle it is, but common code would lead to the least merge conflicts... especially since this file will be changed particularly often

Reply:
Btw, not a directive or anything, just curious what you were thinking :)

Reply:
Yeah, I'm following what we've done for `terraform_version` which is to fail here always in OSS but check it in enterprise. Definitely could lead to merge conflicts in the future, but only if we're modifying code around this if statement.

I think licensing is a different case to here because it's adding a feature that doesn't exist in OSS rather than changing the behavior. There's still probably a good way to utilize build tags, though, I just don't have enough experience to know how without doing some research first. We can discuss these details offline!

For context, the reason I included this configuration in OSS was to prevent merge conflicts in our API representation methods. Our OSS openapi spec needs to include the enterprise-only attributes so that enterprise users can use it as well, and we had the precedence of `terraform_version` in the OSS config, so this seemed like less of a pain than having to resolve the API files. Again, something we can rethink if needed!

Reply:
ok, that sounds good to me

Can see how it goes, and always refactor later if we need to